developer-hub/src/reference_docs/safe_defensive_programming.rs

```diff
@@ -1 +1,615 @@
+//! production/release, they will merely log an error via the logging instance the runtime is using
+//! (i.e., `log::error`).
+//!
+//! This is useful for catching issues in the development environment, without risking panicing in production.
+//!
+//! ## Integer Overflow
+//!
+//! The Rust compiler prevents any sort of static overflow from happening at compile time, for
+//! example:
+//!
+//! ```ignore
+//! let overflow = u8::MAX + 10;
+//! ```
+//! Would yield the following error:
+//! ```text
+//! error: this arithmetic operation will overflow
+//!    --> src/main.rs:121:24
+//!     |
+//! 121 |         let overflow = u8::MAX + 10;
+//!     |                        ^^^^^^^^^^^^ attempt to compute `u8::MAX + 10_u8`, which would overflow
+//!     |
+//! ```
+//!
+//!
+//! However in the runtime context, we don't always have control over what is being supplied as a
+//! parameter. For example, even this simple adding function could present one of two outcomes
+//! depending on whether it is in **release** or **debug** mode:
+#![doc = docify::embed!("./src/reference_docs/safe_defensive_programming.rs", naive_add)]
+//!
+//! If we passed in overflow-able values at runtime, this could actually panic (or wrap, if in
+//! release).
+//! ```ignore
+//! naive_add(250u8, 10u8); // In debug mode, this would panic. In release, this would return 4.
+//! ```
+//!
+//! The Rust compiler would panic in **debug** mode in the event of an integer overflow. In
+//! **release** mode, it resorts to silently _wrapping_ the overflowed amount in a modular fashion,
+//! (hence returning `4`).
+//!
+//! It is actually the _silent_ portion of this behavior that presents a real issue - as it may be
+//! an unintended, but also a very _silent killer_ in terms of producing bugs. In fact, it would
+//! have been better for this type of behavior to produce some sort of error, or even `panic!`, as
+//! in that scenario, at least such behavior could become obvious. Especially in the context of
+//! blockchain development, where unsafe arithmetic could produce unexpected consequences.
+//!
+//! A quick example is a user's balance overflowing: the default behavior of wrapping could result
+//! in the user's balance starting from zero, or vice versa, of a `0` balance turning into the `MAX`
+//! of some type. Naturally, this could lead to various exploits and issues down the road, which if
+//! failing silently, would be difficult to trace and rectify.
+//!
+//! Luckily, there are ways to both represent and handle these scenarios depending on our specific
+//! use case natively built into Rust, as well as libraries like [`sp_arithmetic`].
+//!
+//! ## Infallible Arithmetic
+//!
+//! Our main objective in runtime development is to reduce the likelihood of any *unintended* or
+//! *undefined* behavior. Intentional and predictable design should be our first and foremost
+//! property for ensuring a well running, safely designed system. Both Rust and Substrate both
```

Comment:
We are repeating "both"
```suggestion
//! property for ensuring a well running, safely designed system. Both Rust and Substrate
```